File: StronglyTypedEnumConverter_Tests/InitializationTests.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Shouldly;

namespace StronglyTypedEnumConverter
{
    [TestClass]
    public class InitializationTests
    {
        [TestMethod]
        public void Converter_Initialize_ObjectCreated()
        {
            var converter = new Converter();
            converter.ShouldNotBeNull();
        }


        [TestMethod]
        public void Converter_GarbageIn_ThrowsArgumentException()
        {
            var converter = new Converter();
            try
            {
                converter.Convert("garbage }{");
            }
            catch (ArgumentException ex)
            {
                Console.WriteLine(ex);
                return;
            }

            Assert.Fail("Expected exception did not occur");
        }

    }
}

File: StronglyTypedEnumConverter_VbTests/InitializationTests.vb

Imports Microsoft.VisualStudio.TestTools.UnitTesting


<TestClass> _
Public Class InitializationTests
    <TestMethod> _
    Public Sub Converter_Initialize_ObjectCreated()
        Dim converter = New Converter()
        Assert.IsNotNull(converter)
    End Sub


    <TestMethod> _
    Public Sub Converter_GarbageIn_ThrowsArgumentException()
        Dim converter = New Converter()
        Try
            converter.Convert("End Enum " & vbNewLine & "Public Enum garbage ")
        Catch ex As ArgumentException
            Console.WriteLine(ex)
            Return
        End Try

        Assert.Fail("Expected exception did not occur")
    End Sub

End Class


